Integrate Umami tracking script and implement event tracking

jinhojang6 opened the pull request:
## Umami dashboard
https://umami.bi.status.im/websites/92fb3459-5270-4ce8-a81b-70bee39fbdfe

## Preview
https://contribute-logos-co-git-umami-analytics-acidinfo.vercel.app/en/proposals/

## Todo
Removing the `contribute-logos-co-git-umami-analytics-acidinfo.vercel.app` domain from the script after review

## Screenshot
<img width="1413" height="712" alt="Screenshot 2025-11-26 at 8 10 47 PM" src="https://github.com/user-attachments/assets/d5af5a11-c313-4da4-ab32-ad8e69838d61" />

jkbktl commented on components/form/contact-form.tsx:
```diff
@@ -173,21 +185,28 @@ export default function ContactForm() {
+    window.umami.track('Proposals - Logos Contribute', {
+      source: `${locale}${ROUTES.proposals}/`,
+    })
+
```
should this be logged even if it hasErrors?

jinhojang6 replied:
@jkbktl Thanks for the input and that's a great point, and I had the same thought.

I initially implemented this to test the event tracking of Umami, but I'm open to any suggestions.

I thought the 'intention' to submit the form should be tracked, since the number of actual submissions is already available on the dashboard.

JulesFiliot replied:
I agree with @jinhojang6, I think the goal here is to factually measure how many people tried to submit a proposal. If it fails or succeed is another problem, our goal imo is to see how "active" the form is.

JulesFiliot commented on components/form/contact-form.tsx:
```diff
@@ -4,6 +4,16 @@ import { useState, useRef, useEffect } from 'react'
+      track: (event: string, data: Record<string, string>) => void
+    }
+  }
+}
```
I would suggest moving this declaration [here](https://github.com/acid-info/contribute.logos.co/blob/develop/types/index.d.ts).

jinhojang6 replied:
@JulesFiliot Thanks for the great suggestion. I've moved the declaration

jinhojang6 commented:
Removed the Vercel preview from the data-domains and moved the declaration: https://github.com/acid-info/contribute.logos.co/pull/12/commits/4bc2c3ab5a65ae199ca48f9ab932b6c4027efcd2

jinhojang6 commented:
<img width="1395" height="639" alt="Screenshot 2025-12-11 at 11 06 42 PM" src="https://github.com/user-attachments/assets/b9548bb8-6fb7-48dd-851d-52a635c86ad3" />
Works as expected
